pyupgrade.py

```diff
@@ -2198,10 +2200,10 @@ def _replace(i: int, mapping: Dict[str, str], node: NameOrAttr) -> None:
         elif token.offset in visitor.io_open_calls:
             j = _find_open_paren(tokens, i)
             tokens[i:j] = [token._replace(name='NAME', src='open')]
-        elif token.offset in visitor.mock_mock:
+        elif token.offset in visitor.mock_mock and not no_mock:
```

Comment:
better would be to pass this into the Visitor -- that way we can avoid the expensive tokenization if all that was found to "fix" was mock stuff

Reply:
Sure